### PR TITLE
Use mica in the SUI if mica is enabled in the theme

### PR DESCRIPTION
**Pull request by zadjii-msft**
Does what it says on the tin!

### CODE REVIEW

**Comment by zadjii-msft** [comment]
image
image

File: src/cascadia/TerminalSettingsEditor/MainPage.h [hunk 1]
```diff
@@ -67,6 +67,8 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         void _Navigate(hstring clickedItemTag, BreadcrumbSubPage subPage);
         void _Navigate(const Editor::ProfileViewModel& profile, BreadcrumbSubPage subPage, const bool focusDeleteButton = false);
 
+        void _updateBackgroundForMica();
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.h:70]:
```suggestion
        void _UpdateBackgroundForMica();
```
nit: all the other ones start capitalized

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _settingsClone{ settings.Copy() }
     {
         InitializeComponent();
-
+        _updateBackgroundForMica();
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:58]:
```suggestion
        _UpdateBackgroundForMica();
```
nit: all the other ones start capitalized

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 3]
```diff
@@ -102,6 +102,8 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _settingsSource = settings;
         _settingsClone = settings.Copy();
 
+        _updateBackgroundForMica();
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:105]:
```suggestion
        _UpdateBackgroundForMica();
```
nit: all the other ones start capitalized

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 4]
```diff
@@ -640,4 +642,21 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         return SettingsNav().Background();
     }
 
+    // If the theme asks for Mica, then drop out our background, so that we
+    // can have mica too.
+    void MainPage::_updateBackgroundForMica()
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:647]:
```suggestion
    void MainPage::_UpdateBackgroundForMica()
```
nit: all the other ones start capitalized

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 5]
```diff
@@ -640,4 +642,21 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
     }
 
+    // If the theme asks for Mica, then drop out our background, so that we
+    // can have mica too.
+    void MainPage::_updateBackgroundForMica()
+
+    {
+        const auto& theme = _settingsSource.GlobalSettings().CurrentTheme();
+
+        auto bgKey = (theme.Window() != nullptr && theme.Window().UseMica()) ?
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:652]:
```suggestion
        const auto bgKey = (theme.Window() != nullptr && theme.Window().UseMica()) ?
```

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 6]
```diff
@@ -640,4 +642,21 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
+                         L"SettingsPageMicaBackground" :
+                         L"SettingsPageBackground";
+
+        if (auto bgColor = Resources().TryLookup(winrt::box_value(bgKey)))
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:656]:
```suggestion
        if (const auto bgColor = Resources().TryLookup(winrt::box_value(bgKey)))
```

**Review by carlos-zamora — APPROVED** [review]
I'm down for this. Added all the nits as suggestions so that you can hopefully just apply all of them without opening the branch.

If we end up not liking it later, it should be pretty easy to revert too.

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
Moved from body

---

From discussion yesterday:

mica-in-sui

**Comment by DHowett** [comment]
@msftbot merge this in 1 minute

**Pull request merged**